src/test/kotlin/org/onflow/flow/sdk/impl/AsyncFlowAccessApiImplTest.kt

```diff
@@ -431,4 +432,66 @@ class AsyncFlowAccessApiImplTest {
 
         assertEquals(mockFlowSnapshot, result)
     }
+
+    @Test
+    fun `test getTransactionsByBlockId`() {
+        val blockId = FlowId("01")
+        val transactions = listOf(FlowTransaction.of(TransactionOuterClass.Transaction.getDefaultInstance()))
+
+        val response = Access.TransactionsResponse.newBuilder()
+            .addAllTransactions(transactions.map { it.builder().build() })
+            .build()
+
+        val future: ListenableFuture<Access.TransactionsResponse> = SettableFuture.create()
+        (future as SettableFuture<Access.TransactionsResponse>).set(response)
+
+        `when`(api.getTransactionsByBlockID(any())).thenReturn(future)
+
+        val result = asyncFlowAccessApi.getTransactionsByBlockId(blockId).get()
+
+        assertEquals(transactions, result)
+    }
+
+    @Test
+    fun `test getTransactionResultsByBlockId`() {
+        val blockId = FlowId("01")
+        val transactionResults = listOf(FlowTransactionResult.of(Access.TransactionResultResponse.getDefaultInstance()))
+
+        val response = Access.TransactionResultsResponse.newBuilder()
+            .addAllTransactionResults(transactionResults.map { it.builder().build() })
+            .build()
+
+        val future: ListenableFuture<Access.TransactionResultsResponse> = SettableFuture.create()
+        (future as SettableFuture<Access.TransactionResultsResponse>).set(response)
+
+        `when`(api.getTransactionResultsByBlockID(any())).thenReturn(future)
+
+        val result = asyncFlowAccessApi.getTransactionResultsByBlockId(blockId).get()
+
+        assertEquals(transactionResults, result)
+    }
+
+    @Test
+    fun `test getExecutionResultByBlockId`() {
+        val blockId = FlowId("01")
+        val executionResult = ExecutionResult(FlowId("01"), FlowId("02"))
+
+        val response = Access.ExecutionResultByIDResponse.newBuilder()
+            .setExecutionResult(
+                ExecutionResultOuterClass.ExecutionResult.newBuilder()
+                    .setBlockId(ByteString.copyFrom(byteArrayOf(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1)))
+                    .setPreviousResultId(ByteString.copyFrom(byteArrayOf(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2)))
```

Comment:
Since these get  repeated a bit might be worth refactoring into a shared constant